src/business-logic/services/device-service-implementation.ts

```diff
@@ -0,0 +1,25 @@
+import { DeviceService } from './interfaces/device-service'
+import { Device } from '../../model/entities/device'
+import { DeviceRepository } from '../../data-access/repositories/interfaces/device-repository'
+import { DeviceEventEmitter } from './interfaces/device-event-emitter'
+
+export class DeviceServiceImplementation implements DeviceService {
+  constructor(private readonly deviceRepository: DeviceRepository, private readonly deviceEventEmitter: DeviceEventEmitter) {}
+  
+  public async getDevices(): Promise<Device[]> {
+    return await this.deviceRepository.getDevices()
+  }
+
+  public async getDevice(deviceId: string): Promise<Device> {
+    const deviceConfig = await this.deviceRepository.getDevice(deviceId)
+    if (!deviceConfig) {
+      throw new Error(`Device with id ${deviceId} not found`)
+    }
+    return deviceConfig
+  }
```

Comment:
The "logic" in here is duplicated in `DeviceRepository`. `DeviceRepository.getDevice` throws an exception if it can't find a Device, making the method `DeviceServier.getDevice` redundant.
I suggest removing this method.

Reply:
Passthrough methods allowed after standup-discussion. (Reasons: separation of concerns (the controller doesn't need to know about the DeviceRepository), plus the controller will have to be refactored the minute new business logic is needed anyway - which will happen), plus convention.